airflow/ui/src/components/ClearRun/ClearRunButton.tsx

```diff
@@ -48,7 +56,8 @@ const ClearRunButton = ({ dagId, dagRunId }: Props) => {
 
   return (
     <Box>
-      <Button
+      <ButtonComponent
```

Comment:
We may be able to get away with something like:

```
<Button
  as={withText ? Button : IconButton`
 ```

Reply:
Ah if possible same could be applied to airflow/ui/src/components/TriggerDag/TriggerDAGIconButton.tsx and consolidate both components...

Reply:
> We may be able to get away with something like:
> 
> ```
> <Button
>   as={withText ? undefined : IconButton}
> ```

Good idea, I don't know why the `as IconButton` is not working properly, and the `undefined` part make the pointer and button behavior disappear :(.

Reply:
Maybe someone will find a way to improve it later.

Reply:
> We may be able to get away with something like:
> 
> ```
> <Button
>   as={withText ? undefined : IconButton}
> ```

Good idea but this is not working great. Mostly because the `IconButton` add css (center the icon, modify the background color to transparent or lightgrey i'm not sure), and just defaulting `as` an `IconButton` does not apply this css, rendering is the broken.

I can make this work by overriding `backgroundColor`, `_hover`, and try to manually center, but that's a little bit hacky.

Reply:
>Ah if possible same could be applied to airflow/ui/src/components/TriggerDag/TriggerDAGIconButton.tsx and consolidate both components...

Yes i'll open a follow up PR.

Reply:
Ok yeah ignore my comment. I wasn't sure if it would work